client/src/components/cv/EditCV.tsx

```diff
@@ -163,36 +162,96 @@ function EditCV(props: Props) {
-      desiredPosition,
+      avatarLink: nullifyConditional(avatarLink),
+      desiredPosition: nullifyConditional(desiredPosition),
       englishLevel,
-      name,
-      notes,
-      phone,
-      email,
-      skype,
-      telegram,
-      linkedin,
-      location,
-      githubUsername: github,
-      website,
+      name: nullifyConditional(name),
+      notes: nullifyConditional(notes),
+      phone: nullifyConditional(phone),
+      email: nullifyConditional(email),
+      skype: nullifyConditional(skype),
+      telegram: nullifyConditional(telegram),
+      linkedin: nullifyConditional(linkedin),
+      locations: nullifyConditional(transformLocationsString(locations)),
+      githubUsername: nullifyConditional(github),
+      website: nullifyConditional(website),
       startFrom: startFrom && moment(startFrom).format('YYYY-MM-DD'),
       fullTime,
     };
 
-    await cvService.saveCVData(cvData);
+    const newCVData = await cvService.saveCVData(cvData);
+
+    const {
+      selfIntroLink: newSelfIntroLink,
+      militaryService: newMilitaryService,
+      avatarLink: newAvatarLink,
+      desiredPosition: newDesiredPosition,
+      englishLevel: newEnglishLevel,
+      name: newName,
+      notes: newNotes,
+      startFrom: newStartFrom,
+      fullTime: newFullTime,
+      phone: newPhone,
+      email: newEmail,
+      skype: newSkype,
+      telegram: newTelegram,
+      linkedin: newLinkedin,
+      locations: newLocations,
+      githubUsername: newGithub,
+      website: newWebsite,
+    } = newCVData;
+
+    const newUserData: UserData = {
+      selfIntroLink: newSelfIntroLink,
+      militaryService: newMilitaryService,
+      avatarLink: newAvatarLink,
+      desiredPosition: newDesiredPosition,
+      englishLevel: newEnglishLevel,
+      name: newName,
+      notes: newNotes,
+      startFrom: newStartFrom,
+      fullTime: newFullTime,
+    };
+
+    const newContacts: Contacts = {
+      phone: newPhone,
+      email: newEmail,
+      skype: newSkype,
+      telegram: newTelegram,
+      linkedin: newLinkedin,
+      locations: newLocations,
+      github: newGithub,
+      website: newWebsite,
+    };
+
+    setUserData(newUserData);
+    setContactsList(newContacts);
+
+    setLoading(false);
   };
 
   const resetFields = async () => {
     await fetchData();
   };
 
-  const handleSave = async (data: any) => {
+  const saveData = async (data: any) => {
     await submitData(data);
-    await fetchData();
+  };
+
+  const transformLocationsString = (locationsRaw: string) => {
+    const locations = locationsRaw.split(';');
+    const firstThreeLocations = locations.slice(0, 3);
+    const locationsTrimmed = firstThreeLocations.map(location => location.trim());
+    const locationsStringified = locationsTrimmed.join(';');
+    return locationsStringified;
+  };
+
+  const nullifyConditional = (str: string | null) => {
+    if (typeof str === 'string') return str.replace(/\s/g, '') === '' ? null : str;
```

Comment:
could it be replaced with `str?.trim() === '' ? null : str`

Reply:
done